File: src/main/java/com/ject/studytrip/image/application/event/ImageEventListener.java
package com.ject.studytrip.image.application.event;

import com.ject.studytrip.image.application.service.ImageService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;
import org.springframework.transaction.event.TransactionPhase;
import org.springframework.transaction.event.TransactionalEventListener;

@Component
@RequiredArgsConstructor
public class ImageEventListener {

    private final ImageService imageService;

    @TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
    public void handleCleanupBatch(ImageCleanupBatchEvent event) {
        imageService.cleanupBatch(event.imageUrls());
    }
}

File: src/main/kotlin/com/ject/studytrip/image/application/event/ImageEventListener.kt
package com.ject.studytrip.image.application.event

import com.ject.studytrip.image.application.service.ImageService
import org.springframework.stereotype.Component
import org.springframework.transaction.event.TransactionPhase
import org.springframework.transaction.event.TransactionalEventListener

@Component
class ImageEventListener(
    private val imageService: ImageService,
) {
    @TransactionalEventListener(phase = TransactionPhase.AFTER_COMMIT)
    fun handleCleanupBatch(event: ImageCleanupBatchEvent) = imageService.cleanupBatch(event.imageUrls)
}
